src/mame/drivers/tetrisp2.cpp

```diff
@@ -1860,53 +1839,174 @@ TIMER_DEVICE_CALLBACK_MEMBER(stepstag_state::field_cb)
+	screen->set_raw(xtal, 429, 0, 352, 525, 0, 240);	// should be 0-479 interlaced
 }
 
+void stepstag_state::simulate_pc()
+{
+	/*command list(stepstag):
+	$010x - start mpeg playing?
+	$02xx - ?
+	$03xx - open mpeg file set
+		$0311 jaleco logo animation
+		$0319 demo L/R bg
+	$04xx - ?
+	$07xx - load sound files to particular addresses of ymz280b ram
+		$070x stereo game music
+		$071x sound effects
+		$072x sample set for music selection
+		$07ff clear music sample region?
+	*/
+	FILE *inFile;
+	unsigned int size, position;	// position2
+	const unsigned int total_ram = 0x1000000;	// full 16MB for each ymz280b
+	if(strcmp(machine().system().name,"vjdash") == 0)
+		position = 0x400000;
+	//	position2 = 0xa00000;	// bass?
+	else
+		position = 0x900000;
+
+	if((m_stepstag_main2pc & 0xff0) == 0x700)
+	{
+		inFile = fopen("bgm_l", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram + position, 1, size, inFile);
+			fclose(inFile);
+		}
+//		else
+//			printf("BGM file of Left Channel not found!\n");
+
+		inFile = fopen("bgm_r", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram1 + position, 1, size, inFile);
+			fclose(inFile);
+		}
+//		else
+//			printf("BGM file of Right Channel not found!\n");
+	}
+	else if((m_stepstag_main2pc & 0x0ff0)==0x720)	//music selection
+	{
+		position += 0x100000;
+
+		inFile=fopen("bgm_sample", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram + position, 1, size, inFile);
+			fclose(inFile);
+			memcpy(m_ymzram1 + position, m_ymzram + position, size);
+		}
+	}
+	else if((m_stepstag_main2pc & 0x0fff)==0x714)	//sound effects
+	{
+		position = 0;
+
+		inFile=fopen("sound_effects", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram + position, 1, size, inFile);
+			fclose(inFile);
+			memcpy(m_ymzram1 + position, m_ymzram + position, size);
+		}
+	}
+	else if((m_stepstag_main2pc & 0x0fff)==0x715)	//sound effects 2
+	{
+		position = 0x400000;
+
+		inFile=fopen("vjs02.vjs", "rb");
+		if( inFile )
+		{
+			fseek(inFile, 0, SEEK_END);
+			size = ftell(inFile);
+			rewind(inFile);
+			if(size > total_ram - position)
+				size = total_ram - position;
+			fread(m_ymzram + position, 1, size, inFile);
+			fclose(inFile);
+			memcpy(m_ymzram1 + position, m_ymzram + position, size);
+		}
+	}
+	else if((m_stepstag_main2pc & 0x0f00)==0x7ff)
+	{
+	}
+	else if((m_stepstag_main2pc & 0x0f00)==0x700)	 printf(" cmd 0x%x !! unknown !!!\n", m_stepstag_main2pc);
```

Comment:
Your indentation went wild here.
And anyway: last two branches must be merged into a single generic catch-all `else`, it also belong to logerror again.